### PR TITLE
Update docs with new branches

### PR DESCRIPTION
**Pull request by blueandgold**
Thanks for opening a Pull Request!

Here's a handy checklist to ensure your PR goes smoothly.

- [ ] I signed Google's [Contributor License Agreement](https://opensource.google.com/docs/cla/)
- [ ] My code conforms to Google's [python style](https://google.github.io/styleguide/pyguide.html).
- [ ] My PR at a minimum doesn't decrease unit-test coverage (if applicable).
- [ ] My PR has been functionally tested.
- [ ] All of the [unit-tests](http://forsetisecurity.org/docs/development/#executing-tests) still pass.
- [ ] Running `pylint --rcfile=pylintrc` passes.

These guidelines and more can be found in our [contributing guidelines](https://github.com/GoogleCloudPlatform/forseti-security/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: _docs/_latest/develop/branch-management.md [hunk 1]
```diff
@@ -44,19 +50,29 @@ After the EAP is complete, a Release Candidate (RC) will be created, like
...
+* `3.0-rc2`: A second release candidate with additional changes.
+* `master`: The final released version of v2.0.
 
-* `2.0-dev`: A daily development branch.
-* `2.0-eap1`: A branch used for Early Access testing.
-* `2.0-rc1`: A possible release candidate.
-* `2.0-rc2`: A second release candidate with additional changes.
-* `2.0-master`: The final released version of v2.0.
+Before launching a new version, the existing `dev` and `master` will be moved
+and renamed as `(N-1).0-dev` and `(N-1).0-master`.  Existing `N.N-dev`
```
Comment (mirons-google) [inline _docs/_latest/develop/branch-management.md]: One space after a period, fix everywhere please.

File: _docs/_latest/develop/branch-management.md [hunk 2]
```diff
@@ -44,19 +50,29 @@ After the EAP is complete, a Release Candidate (RC) will be created, like
...
 
-After the support period has passed for the previous version, the
-next-generation `dev` and `master` branches will be merged into the
-`dev` and `master` branches. Then, the `N.N-<name>` branches will be deleted. 
-This is so that the `dev` and `master` are always maintained as the canonical
-branches, and all the commit histories are retained.
+For example, when version 3.0 is ready to be launched:
+
+```
+dev --> 2.0-dev
```
Comment (mirons-google) [inline _docs/_latest/develop/branch-management.md]: This isn't self explanatory, provide commentary with the a `# Comment.` before each line.

File: _docs/_latest/develop/branch-management.md (hunk 1 above)
Comment (blueandgold) [reply _docs/_latest/develop/branch-management.md]: done, old habits die hard.  :)

File: _docs/_latest/develop/branch-management.md (hunk 2 above)
Comment (blueandgold) [reply _docs/_latest/develop/branch-management.md]: how about this?

**Comment by blueandgold** [comment]
Adding `cla: yes` label, as all commit authors have contributed to this project before.

**Pull request merged**